crates/sui-framework/packages/sui-framework/sources/token.move

```diff
@@ -621,48 +608,60 @@ module sui::token {
         cap: &mut TreasuryCap<T>,
         _ctx: &mut TxContext
     ): u64 {
-        let amount = balance::value(&self.spent_balance);
-        let balance = balance::split(&mut self.spent_balance, amount);
-        balance::decrease_supply(coin::supply_mut(cap), balance)
+        let amount = self.spent_balance.value();
+        let balance = self.spent_balance.split(amount);
+        coin::supply_mut(cap).decrease_supply(balance)
     }
 
     // === Getters: `TokenPolicy` and `Token` ===
 
     /// Check whether an action is present in the rules VecMap.
     public fun is_allowed<T>(self: &TokenPolicy<T>, action: &String): bool {
-        vec_map::contains(&self.rules, action)
+        self.rules.contains(action)
     }
 
     /// Returns the rules required for a specific action.
     public fun rules<T>(
         self: &TokenPolicy<T>, action: &String
     ): VecSet<TypeName> {
-        *vec_map::get(&self.rules, action)
+        *self.rules.get(action)
     }
 
     /// Returns the `spent_balance` of the `TokenPolicy`.
     public fun spent_balance<T>(self: &TokenPolicy<T>): u64 {
-        balance::value(&self.spent_balance)
+        self.spent_balance.value()
     }
 
     /// Returns the `balance` of the `Token`.
     public fun value<T>(t: &Token<T>): u64 {
-        balance::value(&t.balance)
+        t.balance.value()
     }
 
     // === Action Names ===
 
     /// Name of the Transfer action.
-    public fun transfer_action(): String { string::utf8(TRANSFER) }
+    public fun transfer_action(): String {
+        let transfer_str = TRANSFER;
+        transfer_str.to_string()
+    }
 
     /// Name of the `Spend` action.
-    public fun spend_action(): String { string::utf8(SPEND) }
+    public fun spend_action(): String {
+        let spend_str = SPEND;
+        spend_str.to_string()
+    }
 
     /// Name of the `ToCoin` action.
-    public fun to_coin_action(): String { string::utf8(TO_COIN) }
+    public fun to_coin_action(): String {
+        let to_coin_str = TO_COIN;
+        to_coin_str.to_string()
```

Comment:
Does this not work with `TO_COIN.to_string()`? Or do we hit the warning for copying a constant? I feel like we shouldn't since its not borrowing it?

Reply:
We hit the copy issue, yeah. I'm not too worried because it does make it clear the constant is being copied. Unfortunately, it actually is borrowing it by lowering it to `TempBorrow` with a `Dereference.` I am happy to revisit the semantics of path resolution if you'd like, since this is the second time this has come up.